testschematic/tests.go

```diff
@@ -17,16 +17,19 @@ import (
 // 3. configure supplied test variables in workspace
 // 4. run PLAN/APPLY/DESTROY steps on workspace to provision and destroy resources
 // 5. delete the test workspace
-func (options *TestSchematicOptions) RunSchematicTest() error {
+
+// create new schematic service with authenticator
+var svc *SchematicsTestService
+
+func (options *TestSchematicOptions) RunSchematicTest() (*SchematicsTestService, error) {
```

Comment:
I'm not sure this change (creating package global for the service pointer and returning it from the Run function) is necessary. The original design was to create a new service object, only if not initially provided in options struct, and then store that pointer in the options struct to be used later if needed. 

The change that is likely needed here would be to make `options.schematicsTestSvc` a public struct attribute by capatalizing the name: `options.SchematicsTestSvc`. By making that attribute public you can then reference it inside this package AND outside in the terraform unit test, and avoid having a package global containing a pointer (which is sometimes avoided due to complexity).

I would remove the global `svc`, remove the return value, and just change the options attribute to public (simply capatalize the attribute name).